src/main/kotlin/com/hibob/academy/dao/PetDao.kt

```diff
@@ -0,0 +1,79 @@
+
+import com.hibob.academy.dao.*
+import jakarta.inject.Inject
+import org.jooq.DSLContext
+import org.jooq.Record
+import org.jooq.RecordMapper
+import org.springframework.stereotype.Component
+import java.sql.Date
+import kotlin.random.Random
+
+@Component
+
+class PetDao @Inject constructor(private val sql: DSLContext) {
+
+    private val pet = PetTable.instance
+    private val companyId = Random.nextLong()
+
+    private val petMapper = RecordMapper<Record, PetData> { record ->
+        PetData(
+            record[pet.name],
+            convertStringToPetType(record[pet.type]),
+            record[pet.companyId],
+            Date(record[pet.dateOfArrival].time)
+        )
+    }
+
+    private val petMapperWithoutType = RecordMapper<Record, PetDataWithoutType> { record ->
+        PetDataWithoutType(
+            record[pet.name],
+            record[pet.companyId],
+            Date(record[pet.dateOfArrival].time)
+        )
+    }
```

Comment:
Why do we need this mapper?
Do we have `Pet` without a `petType`?

Reply:
yes, because there is a function where we get all pets by type and there we will return the data without the type and there is a function where we get all pets and there we do want the types